core/node/rpc/sync/client/local.go

```diff
@@ -0,0 +1,152 @@
+package client
+
+import (
+	"context"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	. "github.com/river-build/river/core/node/base"
+	"github.com/river-build/river/core/node/events"
+	. "github.com/river-build/river/core/node/protocol"
+	. "github.com/river-build/river/core/node/shared"
+)
+
+type localSyncer struct {
+	syncStreamCtx    context.Context
+	syncStreamCancel context.CancelFunc
+
+	streamCache events.StreamCache
+	cookies     []*SyncCookie
+	messages    chan<- *SyncStreamsResponse
+	localAddr   common.Address
+
+	activeStreamsMu sync.Mutex
+	activeStreams   map[StreamId]events.SyncStream
+}
+
+func newLocalSyncer(
+	ctx context.Context,
+	localAddr common.Address,
+	streamCache events.StreamCache,
+	cookies []*SyncCookie,
+	messages chan<- *SyncStreamsResponse,
+) (*localSyncer, error) {
+	syncStreamCtx, syncStreamCancel := context.WithCancel(ctx)
```

Comment:
is this cancel ever called?

(parent context gets canceled as well, is there a need for separate cancelable context here?)